fairmq/shmem/Socket.h

```diff
@@ -126,6 +115,24 @@ class Socket final : public fair::mq::Socket
         return zmq::Connect(fSocket, address, fId);
     }
 
+  private:
+    __FAIRMQ_ALWAYS_INLINE auto MakeMetaMsg(shmem::Message const& shmMsg) noexcept
+    {
+        if (fMetadataMsgSize > sizeof(MetaHeader)) {
+            zmq::UMsg metaMsg(fMetadataMsgSize);
```

Comment:
This should probably also be a `ZMsg` to avoid a copy.

Reply:
Yeah. Also in the vector case?